docs/non-normative/how-to-write-conventions/resource-and-entities.md

```diff
@@ -0,0 +1,301 @@
+  - [How to define identifying attributes](#how-to-define-identifying-attributes)
+  - [How to namespace entities](#how-to-namespace-entities)
+- [Background](#background-resource-and-entities)
+
+## Modelling Guide
+
+To define an entity, create a new semantic convention model file, with a group type as `entity`, for example:
+
+`model/{my_domain}/entities.yaml`:
+
+```yaml
+groups:
+  - id: entity.my_entity
+    type: entity
+    stability: development
+    name: my_entity
+    brief: >
+      A description of my_entity here.
+    attributes:
+      - ref: some.attribute
+        role: identifying
+      - ref: some.other_attribute
+        role: descriptive
+        ...
+```
+
+Here, the attributes field contains all attributes of the Entity.
+The `role` of each attribute determines if it is identifying or
+descriptive. See
+[How to define identifying attributes?](#how-to-define-identifying-attributes)
+for details on what these mean.
+
+> Note: Declaring Entity relationships is not yet supported.
+
+### Declaring associations between signals
+
+You can declare which entities should be used with specific observability
+signals.  For example, process metrics should be used with the process
+entity, so that the metric is associated with a known process. To declare
+this, use the `entity_associations` field on the signal and reference
+another resource group *by name*.
+
+`model/{my_domain}/metrics.yaml`:
+
+```yaml
+groups:
+  - id: metric.some_metric
+    type: metric
+    ...
+    entity_associations: 
+      - my_entity
+```
+
+Notes:
+
+- You cannot declare an association on an *unstable* resource from a
+  *stable* signal.
+- You can declare multiple associations. These form a "one or many" set,
+  where one or many of the named entities may be associated with the
+  metric. There is *no* requirement on disjointness due to how is-a
+  relationships are modelled.
+
+### Extending an entity
+
+While not recommended for Semantic Conventions, you can define a new
+"view" of an entity that includes additional descriptive attributes. To
+do so, use the extends field on groups:
+
+`model/{my_other_domain}/entities.yaml`:
+
+```yaml
+groups:
+  - id: entity.my_entity_2
+    type: entity
+    extends: entity.my_entity
+    attributes:
+      - ref: new.attribute.name
+        requirement_level: opt_in
+        role: descriptive
+```
+
+Notes:
+
+- You cannot change the `name` or `type` fields of the new entity.
+- You cannot change the set of identifying attributes.
+
+## FAQ
+
+### When to define a new entity?
+
+There are two scenarios where entities should be defined:
+
+- When you are generating a new signal (log, metric, span, etc.)
+  and no existing entity makes sense as the "source".
+- (future) When you need to describe an entity hierarchy from
+  some system of record (e.g. resources in kubernetes, assets
+  in a cloud).
+
+For example, the Mainframe SIG in OpenTelemetry will need to outline a
```

Comment:
The example is not clear, I don't think it helps. I don't know what Mainframe SIG plans to do and I think vast majority of this document readers don't know either. Is it possible to include an example from a more well-known domain?